### PR TITLE
Optimize DCS input parameters

### PR DESCRIPTION
**Pull request by smilemusic**
$ make testacc TEST=./flexibleengine/ TESTARGS='-run TestAccDcsInstancesV1_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./flexibleengine/ -v -run TestAccDcsInstancesV1_basic -timeout 720m
=== RUN   TestAccDcsInstancesV1_basic
--- PASS: TestAccDcsInstancesV1_basic (540.45s)
PASS
ok      github.com/terraform-providers/terraform-provider-flexibleengine/flexibleengine 540.655s

### CODE REVIEW

File: flexibleengine/resource_flexibleengine_dcs_instance_v1.go [hunk 1]
```diff
@@ -85,7 +86,7 @@ func resourceDcsInstanceV1() *schema.Resource {
 				ForceNew: true,
 				Elem:     &schema.Schema{Type: schema.TypeString},
 			},
-			"product_id": {
+			"instance_type": {
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dcs_instance_v1.go:89]: we should add a new argument to keep the compatibility.

File: flexibleengine/resource_flexibleengine_dcs_instance_v1.go [hunk 2]
```diff
@@ -196,7 +197,45 @@ func getInstanceBackupPolicy(d *schema.ResourceData) *instances.InstanceBackupPo
 	return instanceBackupPolicy
 }
 
+func getDcsProductId(d *schema.ResourceData, meta interface{}) error {
```
Comment (ShiChangkuo) [inline flexibleengine/resource_flexibleengine_dcs_instance_v1.go:200]: no need to get the client twice, we can define the function as: ```func getDcsProductId(client *golangsdk.ServiceClient, instanceType string) (string, error) { ... }```

Comment (smilemusic) [reply flexibleengine/resource_flexibleengine_dcs_instance_v1.go:200]: ok

Comment (smilemusic) [reply flexibleengine/resource_flexibleengine_dcs_instance_v1.go:200]: > no need to get the client twice, we can define the function as: `func getDcsProductId(client *golangsdk.ServiceClient, instanceType string) (string, error) { ... }`

ok

**Pull request merged**